File: JAVA/19_Queue,Deque/BOJ_01_10845_큐.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.Deque;
import java.util.ArrayDeque;

public class Main {
    public static void main(String[] args) throws IOException {
        Deque<Integer> q = new ArrayDeque<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            String str = st.nextToken();
            if (str.contains("push")) {
                q.offer(Integer.parseInt(st.nextToken()));
            } else if (str.contains("pop")) {
                Integer p = q.poll();
                if (p == null) {
                    sb.append(-1).append('\n');
                } else {
                    sb.append(p).append('\n');
                }
            } else if (str.contains("size"))
                sb.append(q.size()).append('\n');
            else if (str.contains("empty")) {
                if (q.isEmpty()) {
                    sb.append(1).append('\n');
                } else {
                    sb.append(0).append('\n');
                }
            } else if (str.contains("front")) {
                Integer f = q.peek();
                if (f == null) {
                    sb.append(-1).append('\n');
                } else {
                    sb.append(f).append('\n');
                }
            } else if (str.contains("back")) {
                Integer r = q.peekLast();
                if (r == null) {
                    sb.append(-1).append('\n');
                } else {
                    sb.append(r).append('\n');
                }
            }
        }
        System.out.println(sb);
    }
}

File: C/19_Queue,Deque/BOJ_01_10845_큐.c
#include <stdio.h>
int queue[10000];
int f = 0, r = 0;
void push(int x) {
	queue[r++] = x;
}
int pop() {
	if (r == f) return -1;
	else return queue[f++];
}
int size() {
	return (r - f);
}
int empty() {
	if (r == f) return 1;
	else return 0;
}
int front() {
	if (r == f) return -1;
	else return queue[f];
}
int back() {
	if (r == f) return -1;
	else return queue[r - 1];
}

main() {
	int n, cnt = 0;
	char order[6];
	scanf("%d", &n);
	int* result = (int*)malloc(sizeof(int) * n);
	for (int i = 0; i < n; i++) {
		scanf("%s", &order);
		if (order[0] == 'p') {
			if (order[1] == 'u') {
				int x;
				scanf("%d", &x);
				push(x);
			}
			else result[cnt++] = pop();
		}
		else if (order[0] == 's') {
			result[cnt++] = size();
		}
		else if (order[0] == 'e') {
			result[cnt++] = empty();
		}
		else if (order[0] == 'f') {
			result[cnt++] = front();
		}
		else if (order[0] == 'b') {
			result[cnt++] = back();
		}
	}
	for (int i = 0; i < cnt; i++) {
		printf("%d\n", result[i]);
	}
}